numba_cuda/numba/cuda/tests/cudadrv/test_nvjitlink.py

```diff
@@ -251,5 +260,74 @@ def kernel():
                 pass
 
 
+@unittest.skipIf(
+    not PYNVJITLINK_INSTALLED, reason="Pynvjitlink is not installed"
+)
+class TestLinkerUsage(CUDATestCase):
+    """Test that whether pynvjitlink can be enabled by both environment variable
+    and modification of config at runtime.
+    """
+
+    src = """if 1:
+        import os
+        from numba import cuda
+
+        TEST_BIN_DIR = os.getenv("NUMBA_CUDA_TEST_BIN_DIR")
+        if TEST_BIN_DIR:
+            test_device_functions_cubin = os.path.join(
+                TEST_BIN_DIR, "test_device_functions.cubin"
+            )
+
+        sig = "uint32(uint32, uint32)"
+        add_from_numba = cuda.declare_device("add_from_numba", sig)
+
+        @cuda.jit(link=[test_device_functions_cubin], lto=True)
+        def kernel(result):
+            result[0] = add_from_numba(1, 2)
+
+        result = cuda.device_array(1)
+        kernel[1, 1](result)
+        assert result[0] == 3
+        """
+
+    def test_linker_enabled_envvar(self):
+        env = os.environ.copy()
+        env['NUMBA_CUDA_ENABLE_PYNVJITLINK'] = "1"
+        run_in_subprocess(self.src, env=env)
+
+    def test_linker_disabled_envvar(self):
+        env = os.environ.copy()
+        env.pop('NUMBA_CUDA_ENABLE_PYNVJITLINK', None)
+        with self.assertRaisesRegex(
+            AssertionError, "LTO and additional flags require PyNvJitLinker"
+        ):
+            # Actual error raised is `ValueError`, but `run_in_subprocess`
+            # reraises as AssertionError.
+            run_in_subprocess(self.src, env=env)
+
+    def _lto_test(self):
+        sig = "uint32(uint32, uint32)"
+        add_from_numba = cuda.declare_device("add_from_numba", sig)
+
+        @cuda.jit(link=[test_device_functions_cubin], lto=True)
+        def kernel(result):
+            result[0] = add_from_numba(1, 2)
+
+        result = cuda.device_array(1)
+        kernel[1, 1](result)
+        assert result[0] == 3
+
+    def test_linker_enabled_config(self):
+        with override_config("CUDA_ENABLE_PYNVJITLINK", True):
```

Comment:
With the latest changes, I still think this test needs to run in a subprocess without pynvjitlink enabled by an environment variable - in its present form in CI, this test is running with pynvjitlink enabled by the environment variable, so even if setting the config variable has no effect I think it would still pass.

(If I'm misunderstanding the flow here, please point me in the right direction to understand what's going on :-) )

Reply:
Ok yeah, you are right on this. I think the test should reflect what we are actually trying to achieve - in a separate process, can user use pynvjitlink with numba-cuda by just turning on the config item at runtime? We do need to run it in separate process to make sure Numba is loaded from scratch.